### PR TITLE
update minio-go to latest master

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

**Comment by harshavardhana** [comment]
PTAL @klauspost at the earliest, I need this for a customer deployment tomorrow my time.

File: pkg/bench/get.go [hunk 1]
```diff
@@ -186,8 +186,12 @@ func (g *Get) Start(ctx context.Context, wait chan struct{}) (Operations, error)
 				opts.VersionID = obj.VersionID
 				o, err := client.GetObject(ctx, g.Bucket, obj.Name, opts)
 				if err != nil {
-					console.Errorln("download error:", err)
-					op.Err = err.Error()
+					// In-flight requests may get canceled due to
+					// auto-term or reaching the `--duration` specified
+					if !isErrContext(err) {
+						console.Errorln("download error:", err)
+						op.Err = err.Error()
```
Comment (klauspost) [inline pkg/bench/get.go:193]: This is wrong. It will be counted as a completed request which will skew numbers.

Switch to a non-terminating context like the others.  I will make a PR.

File: pkg/bench/mixed.go [hunk 2]
```diff
@@ -268,8 +268,12 @@ func (g *Mixed) Start(ctx context.Context, wait chan struct{}) (Operations, erro
 					o, err := client.GetObject(nonTerm, g.Bucket, obj.Name, getOpts)
 					fbr.r = o
 					if err != nil {
-						console.Errorln("download error:", err)
-						op.Err = err.Error()
+						// In-flight requests may get canceled due to
```
Comment (klauspost) [inline pkg/bench/mixed.go:271]: No, the nonterminating context is used.

File: pkg/bench/mixed.go [hunk 3]
```diff
@@ -288,8 +301,8 @@ func (g *Mixed) Start(ctx context.Context, wait chan struct{}) (Operations, erro
 						console.Errorln(op.Err)
 					}
 					rcv <- op
-					objDone()
 					clDone()
+					objDone()
```
Comment (klauspost) [inline pkg/bench/mixed.go:305]: What is the reason for this change?

Comment (harshavardhana) [reply pkg/bench/mixed.go:305]: a clDone() should have a objDone otherwise we are leaking something here ?

**Comment by harshavardhana** [comment]
Need minio-go updates as well from this PR for the changes in minio-go

**Comment by harshavardhana** [comment]
I updated the PR title to only include minio-go updates @klauspost PTAL
